-1

Tested at: 4c2a866191d29471dcc3de31be6998863bdd069d


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/5387f33035f82196d51968058b52bc670f79533c
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19023/21062/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19023/21062/git-merge-result

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19023/21062/summary.html

I found follow errors while testing this PR

**Failed tests:  Build  ClangBuild**

* **Build**:

I found an error when building: 
 
 <pre>>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-02-0000/src/DQMOffline/Trigger/plugins/NoBPTXMonitor.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-02-0000/src/DQMOffline/Trigger/plugins/HLTTagAndProbeOfflineSource.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-02-0000/src/DQMOffline/Trigger/plugins/PhotonMonitor.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-02-0000/src/DQMOffline/Trigger/plugins/RazorMonitor.cc 
In file included from /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-02-0000/src/DQMOffline/Trigger/plugins/PhotonMonitor.cc:1:0:
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-02-0000/src/DQMOffline/Trigger/plugins/PhotonMonitor.h:109:12: error: redeclaration of 'PhotonME PhotonMonitor::photonEtaME_'
   PhotonME photonEtaME_;
            ^
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-02-0000/src/DQMOffline/Trigger/plugins/PhotonMonitor.h:107:12: note: previous declaration 'PhotonME PhotonMonitor::photonEtaME_'
   PhotonME photonEtaME_;
            ^
</pre>

* **Clang**:

I found a compilation error while trying to compile with clang: 
I used this command: 
 scram b vclean && scram build -k -j 16 USER_CXXFLAGS='-fsyntax-only' COMPILER='llvm compile'


 <pre>>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-02-0000/src/DQMOffline/Trigger/plugins/HLTTagAndProbeOfflineSource.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-02-0000/src/DQMOffline/Trigger/plugins/PhotonMonitor.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-02-0000/src/DQMOffline/Trigger/plugins/RazorMonitor.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-02-0000/src/DQMOffline/Trigger/plugins/DQMGenericTnPClient.cc 
In file included from /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-02-0000/src/DQMOffline/Trigger/plugins/PhotonMonitor.cc:1:
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-02-0000/src/DQMOffline/Trigger/plugins/PhotonMonitor.h:109:12: error: duplicate member 'photonEtaME_'
  PhotonME photonEtaME_;
           ^
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-02-0000/src/DQMOffline/Trigger/plugins/PhotonMonitor.h:107:12: note: previous declaration is here
  PhotonME photonEtaME_;
           ^
</pre>


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/5387f33035f82196d51968058b52bc670f79533c
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19023/21062/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19023/21062/git-merge-result

--- a/DQMOffline/Trigger/plugins/PhotonMonitor.cc
+++ b/DQMOffline/Trigger/plugins/PhotonMonitor.cc
@@ -17,6 +17,7 @@ PhotonMonitor::PhotonMonitor( const edm::ParameterSet& iConfig ) :
   , eleToken_             ( mayConsume<reco::GsfElectronCollection>(iConfig.getParameter<edm::InputTag>("electrons") ) )   
   , photonToken_             ( mayConsume<reco::PhotonCollection>      (iConfig.getParameter<edm::InputTag>("photons")      ) )   
   , photon_variable_binning_ ( iConfig.getParameter<edm::ParameterSet>("histoPSet").getParameter<std::vector<double> >("photonBinning") )
+  , diphoton_mass_binning_ ( iConfig.getParameter<edm::ParameterSet>("histoPSet").getParameter<std::vector<double> >("massBinning") )
   , photon_binning_          ( getHistoPSet   (iConfig.getParameter<edm::ParameterSet>("histoPSet").getParameter<edm::ParameterSet>   ("photonPSet")    ) )
   , ls_binning_           ( getHistoLSPSet (iConfig.getParameter<edm::ParameterSet>("histoPSet").getParameter<edm::ParameterSet>   ("lsPSet")     ) )
   , num_genTriggerEventFlag_(new GenericTriggerEventFlag(iConfig.getParameter<edm::ParameterSet>("numGenericTriggerEventPSet"),consumesCollector(), *this))
@@ -47,15 +48,27 @@ PhotonMonitor::PhotonMonitor( const edm::ParameterSet& iConfig ) :
   photonHoverEME_.numerator   = nullptr;       
   photonHoverEME_.denominator = nullptr;
 
-  
-  
+  diphotonMassME_.numerator   = nullptr;
+  diphotonMassME_.denominator = nullptr;
+
+  subphotonME_.numerator   = nullptr;
+  subphotonME_.denominator = nullptr;
+  subphotonME_variableBinning_.numerator   = nullptr;
+  subphotonME_variableBinning_.denominator = nullptr;
+  subphotonEtaME_.numerator   = nullptr;
+  subphotonEtaME_.denominator = nullptr;
+  subphotonPhiME_.numerator   = nullptr;
+  subphotonPhiME_.denominator = nullptr;
+  subphotonEtaPhiME_.numerator   = nullptr;       
+  subphotonEtaPhiME_.denominator = nullptr;
+  subphotonr9ME_.numerator   = nullptr;       
+  subphotonr9ME_.denominator = nullptr;
+  subphotonHoverEME_.numerator   = nullptr;       
+  subphotonHoverEME_.denominator = nullptr;
   
 }
-
-
 PhotonMonitor::~PhotonMonitor()
 {
-  
 }
 
 MEbinning PhotonMonitor::getHistoPSet(edm::ParameterSet const& pset)
@@ -143,7 +156,7 @@ void PhotonMonitor::bookHistograms(DQMStore::IBooker     & ibooker,
   histname = "photonVsLS"; histtitle = "photon pt vs LS";
   bookME(ibooker,photonVsLS_,histname,histtitle,ls_binning_.nbins, ls_binning_.xmin, ls_binning_.xmax,photon_binning_.xmin, photon_binning_.xmax);
   setTitle(photonVsLS_,"LS","Photon pT [GeV]");
-
+  
   histname = "photon_phi"; histtitle = "Photon phi";
   bookME(ibooker,photonPhiME_,histname,histtitle, phi_binning_1.nbins, phi_binning_1.xmin, phi_binning_1.xmax);
   setTitle(photonPhiME_,"Photon #phi","events / 0.1 rad");
@@ -152,7 +165,6 @@ void PhotonMonitor::bookHistograms(DQMStore::IBooker     & ibooker,
   histname = "photon_eta"; histtitle = "Photon eta";
   bookME(ibooker,photonEtaME_,histname,histtitle, eta_binning_.nbins, eta_binning_.xmin,eta_binning_.xmax);
   setTitle(photonEtaME_,"Photon #eta","events");
-
 
   histname = "photon_r9"; histtitle = "Photon r9";
   bookME(ibooker,photonr9ME_,histname,histtitle, r9_binning_.nbins, r9_binning_.xmin, r9_binning_.xmax);
@@ -167,6 +179,37 @@ void PhotonMonitor::bookHistograms(DQMStore::IBooker     & ibooker,
   bookME(ibooker,photonEtaPhiME_,histname,histtitle, eta_binning_.nbins, eta_binning_.xmin, eta_binning_.xmax,phi_binning_1.nbins, phi_binning_1.xmin, phi_binning_1.xmax);
   setTitle(photonEtaPhiME_,"#eta","#phi"); 
 
+  //for diphotons
+  if(nphotons_>1)
+    {
+      histname = "diphoton_mass"; histtitle = "Diphoton mass";
+      bookME(ibooker,diphotonMassME_,histname,histtitle, diphoton_mass_binning_);
+      setTitle(diphotonMassME_,"Diphoton mass","events / 0.1");
+      
+      histname = "subphoton_pt"; histtitle = "subphoton PT";
+      bookME(ibooker,subphotonME_,histname,histtitle,photon_binning_.nbins,photon_binning_.xmin, photon_binning_.xmax);
+      setTitle(subphotonME_,"subPhoton pT [GeV]","events / [GeV]");
+      
+      histname = "subphoton_eta"; histtitle = "subPhoton eta";
+      bookME(ibooker,subphotonEtaME_,histname,histtitle, eta_binning_.nbins, eta_binning_.xmin,eta_binning_.xmax);
+      setTitle(subphotonEtaME_,"subPhoton #eta","events / 0.1");
+      
+      histname = "subphoton_phi"; histtitle = "subPhoton phi";
+      bookME(ibooker,subphotonPhiME_,histname,histtitle, phi_binning_1.nbins, phi_binning_1.xmin, phi_binning_1.xmax);
+      setTitle(subphotonPhiME_,"subPhoton #phi","events / 0.1 rad");
+      
+      histname = "subphoton_r9"; histtitle = "subPhoton r9";
+      bookME(ibooker,subphotonr9ME_,histname,histtitle, r9_binning_.nbins, r9_binning_.xmin, r9_binning_.xmax);
+      setTitle(subphotonr9ME_,"subPhoton r9","events");
+      
+      histname = "subphoton_hoE"; histtitle = "subPhoton hoverE";
+      bookME(ibooker,subphotonHoverEME_,histname,histtitle, hoe_binning_.nbins, hoe_binning_.xmin, hoe_binning_.xmax);
+      setTitle(subphotonHoverEME_,"subPhoton hoE","events");
+      
+      histname = "subphoton_etaphi"; histtitle = "subPhoton eta-phi"; 
+      bookME(ibooker,subphotonEtaPhiME_,histname,histtitle, eta_binning_.nbins, eta_binning_.xmin, eta_binning_.xmax,phi_binning_1.nbins, phi_binning_1.xmin, phi_binning_1.xmax);
+      setTitle(subphotonEtaPhiME_,"#eta","#phi"); 
+    }
   // Initialize the GenericTriggerEventFlag
   if ( num_genTriggerEventFlag_ && num_genTriggerEventFlag_->on() ) num_genTriggerEventFlag_->initRun( iRun, iSetup );
   if ( den_genTriggerEventFlag_ && den_genTriggerEventFlag_->on() ) den_genTriggerEventFlag_->initRun( iRun, iSetup );
@@ -209,12 +252,11 @@ void PhotonMonitor::analyze(edm::Event const& iEvent, edm::EventSetup const& iSe
   }
   if ( electrons.size() < nelectrons_ ) return;
 
-
-
   edm::Handle<reco::PhotonCollection> photonHandle;
   iEvent.getByToken( photonToken_, photonHandle );
   std::vector<reco::Photon> photons;
   photons.clear();
+  
   if ( photonHandle->size() < nphotons_ ) return;
   for ( auto const & p : *photonHandle ) {
     if ( photonSelection_( p ) ) photons.push_back(p);
@@ -236,7 +278,19 @@ void PhotonMonitor::analyze(edm::Event const& iEvent, edm::EventSetup const& iSe
 	  photonr9ME_.denominator->Fill(photons[0].r9());
 	  photonHoverEME_.denominator->Fill(photons[0].hadTowOverEm());
 	}
-      
+  
+  if(nphotons_>1) 
+    //filling diphoton histograms
+    {
+      subphotonME_.denominator -> Fill(photons[1].pt());
+      subphotonEtaME_.denominator -> Fill(photons[1].eta());
+      subphotonPhiME_.denominator -> Fill(photons[1].phi());
+      subphotonEtaPhiME_.denominator -> Fill(photons[1].eta(), photons[1].phi());
+      subphotonr9ME_.denominator->Fill(photons[1].r9());
+      subphotonHoverEME_.denominator->Fill(photons[1].hadTowOverEm());
+      diphotonMassME_.denominator -> Fill(sqrt(2*photons[0].pt()*photons[1].pt()*(cosh(photons[0].eta()-photons[1].eta())-cos(photons[0].phi()-photons[1].phi()))));
+    }
+
   // applying selection for numerator
   if (num_genTriggerEventFlag_->on() && ! num_genTriggerEventFlag_->accept( iEvent, iSetup) ) return;
 
@@ -251,10 +305,18 @@ void PhotonMonitor::analyze(edm::Event const& iEvent, edm::EventSetup const& iSe
       photonEtaPhiME_.numerator -> Fill(photons[0].eta(), photons[0].phi());
       photonr9ME_.numerator->Fill(photons[0].r9());
       photonHoverEME_.numerator->Fill(photons[0].hadTowOverEm());
-      
-      
     }
-
+  if(nphotons_>1) 
+    //filling diphoton histograms
+    {
+      subphotonME_.numerator -> Fill(photons[1].pt());
+      subphotonEtaME_.numerator -> Fill(photons[1].eta());
+      subphotonPhiME_.numerator -> Fill(photons[1].phi());
+      subphotonEtaPhiME_.numerator -> Fill(photons[1].eta(), photons[1].phi());
+      subphotonr9ME_.numerator->Fill(photons[1].r9());
+      subphotonHoverEME_.numerator->Fill(photons[1].hadTowOverEm());
+      diphotonMassME_.numerator -> Fill(sqrt(2*photons[0].pt()*photons[1].pt()*(cosh(photons[0].eta()-photons[1].eta())-cos(photons[0].phi()-photons[1].phi()))));
+    }
 }
 
 void PhotonMonitor::fillHistoPSetDescription(edm::ParameterSetDescription & pset)
@@ -309,7 +371,8 @@ void PhotonMonitor::fillDescriptions(edm::ConfigurationDescriptions & descriptio
   histoPSet.add<edm::ParameterSetDescription>("photonPSet", metPSet);
   std::vector<double> bins = {0.,20.,40.,60.,80.,90.,100.,110.,120.,130.,140.,150.,160.,170.,180.,190.,200.,220.,240.,260.,280.,300.,350.,400.,450.,1000.};
   histoPSet.add<std::vector<double> >("photonBinning", bins);
-
+  std::vector<double> massbins = {90.,91.,92.,93.,94.,95.,96.,97.,98.,99.,100.,101.,102.,103.,104.,105.,106.,107.,108.,109.,110.,115.,120.,130.,150.,200.};
+  histoPSet.add<std::vector<double> >("massBinning", massbins);
   edm::ParameterSetDescription lsPSet;
   fillHistoLSPSetDescription(lsPSet);
   histoPSet.add<edm::ParameterSetDescription>("lsPSet", lsPSet);

--- a/DQMOffline/Trigger/plugins/PhotonMonitor.h
+++ b/DQMOffline/Trigger/plugins/PhotonMonitor.h
@@ -4,7 +4,6 @@
 #include <string>
 #include <vector>
 
-
 #include "FWCore/Utilities/interface/EDGetToken.h"
 #include "FWCore/Framework/interface/Frameworkfwd.h"
 #include "FWCore/Framework/interface/Event.h"
@@ -61,7 +60,6 @@ public:
   static void fillDescriptions(edm::ConfigurationDescriptions & descriptions);
   static void fillHistoPSetDescription(edm::ParameterSetDescription & pset);
   static void fillHistoLSPSetDescription(edm::ParameterSetDescription & pset);
-
 protected:
 
   void bookHistograms(DQMStore::IBooker &, edm::Run const &, edm::EventSetup const &) override;
@@ -90,11 +88,22 @@ private:
 
 
   std::vector<double> photon_variable_binning_;
+  std::vector<double> diphoton_mass_binning_;
+  
   MEbinning           photon_binning_;
   MEbinning           ls_binning_;
+  
+  PhotonME subphotonEtaME_;
+  PhotonME subphotonME_;
+  PhotonME subphotonPhiME_;
+  PhotonME subphotonME_variableBinning_;
+  PhotonME subphotonEtaPhiME_;   
+  PhotonME subphotonr9ME_;   
+  PhotonME subphotonHoverEME_;   
+  PhotonME diphotonMassME_;
 
-  PhotonME photonME_;
   PhotonME photonEtaME_;
+  PhotonME photonME_;
   PhotonME photonPhiME_;
   PhotonME photonME_variableBinning_;
   PhotonME photonVsLS_;
@@ -142,7 +151,6 @@ private:
   std::unique_ptr<GenericTriggerEventFlag> num_genTriggerEventFlag_ ;
   std::unique_ptr<GenericTriggerEventFlag> den_genTriggerEventFlag_ ;
 
-
   StringCutObjectSelector<reco::MET,true>         metSelection_;
   StringCutObjectSelector<reco::PFJet,true   >    jetSelection_;
   StringCutObjectSelector<reco::GsfElectron,true> eleSelection_;
@@ -150,7 +158,6 @@ private:
   unsigned int njets_;
   unsigned int nphotons_;
   unsigned int nelectrons_;
-
 
 };
 

--- a/DQMOffline/Trigger/python/ExoticaMonitoring_Client_cff.py
+++ b/DQMOffline/Trigger/python/ExoticaMonitoring_Client_cff.py
@@ -35,7 +35,6 @@ photonEfficiency = DQMEDHarvester("DQMGenericClient",
   
 )
 
-
 NoBPTXEfficiency = DQMEDHarvester("DQMGenericClient",
     subDirs        = cms.untracked.vstring("HLT/NoBPTX/*"),
     verbose        = cms.untracked.uint32(0), # Set to 2 for all messages
@@ -58,5 +57,5 @@ NoBPTXEfficiency = DQMEDHarvester("DQMGenericClient",
 exoticaClient = cms.Sequence(
     metEfficiency
     + NoBPTXEfficiency
-   + photonEfficiency
+    + photonEfficiency
 )

--- a/DQMOffline/Trigger/python/ExoticaMonitoring_cff.py
+++ b/DQMOffline/Trigger/python/ExoticaMonitoring_cff.py
@@ -7,5 +7,5 @@ from DQMOffline.Trigger.NoBPTXMonitor_cff import *
 exoticaMonitorHLT = cms.Sequence(
     exoHLTMETmonitoring
     + exoHLTNoBPTXmonitoring
-   + exoHLTPhotonmonitoring
+    + exoHLTPhotonmonitoring
 )

--- a/DQMOffline/Trigger/python/HiggsMonitoring_Client_cff.py
+++ b/DQMOffline/Trigger/python/HiggsMonitoring_Client_cff.py
@@ -1,7 +1,35 @@
 import FWCore.ParameterSet.Config as cms
+from DQMServices.Core.DQMEDHarvester import DQMEDHarvester
 
 from DQMOffline.Trigger.VBFMETMonitor_Client_cff import *
 
+diphotonEfficiency = DQMEDHarvester("DQMGenericClient",
+    subDirs        = cms.untracked.vstring("HLT/photon/HLT_Diphoton30_22_R9Id_OR_IsoCaloId_AND_HE_R9Id_Mass90_v","HLT/photon/HLT_Diphoton30_22_R9Id_OR_IsoCaloId_AND_HE_R9Id_Mass95_v","HLT_Diphoton30PV_18PV_R9Id_AND_IsoCaloId_AND_HE_R9Id_PixelVeto_Mass55_v","HLT_Diphoton30PV_18PV_R9Id_AND_IsoCaloId_AND_HE_R9Id_NoPixelVeto_Mass55_v","HLT_Diphoton30EB_18EB_R9Id_OR_IsoCaloId_AND_HE_R9Id_NoPixelVeto_Mass55_v","HLT_Diphoton30EB_18EB_R9Id_OR_IsoCaloId_AND_HE_R9Id_PixelVeto_Mass55_v"),
+                                    #subDirs        = cms.untracked.vstring("HLT/Higgs/*"),
+    verbose        = cms.untracked.uint32(0), # Set to 2 for all messages
+    resolution     = cms.vstring(),
+    efficiency     = cms.vstring(
+        "eff_diphoton_pt       'efficiency vs lead pt;             Photon pt [GeV]; efficiency'     photon_pt_numerator          photon_pt_denominator",
+        "eff_diphoton_variable 'efficiency vs lead pt;             Photon pt [GeV]; efficiency'     photon_pt_variable_numerator photon_pt_variable_denominator",
+        "eff_diphoton_eta      'efficiency vs lead eta;            Photon eta; efficiency'          photon_eta_numerator         photon_eta_denominator",
+        "eff_diphoton_subpt    'efficiency vs sublead pt;          Photon subpt [GeV]; efficiency'  subphoton_pt_numerator       subphoton_pt_denominator",
+        "eff_diphoton_subeta   'efficiency vs sublead eta;         Photon subeta; efficiency'       subphoton_eta_numerator      subphoton_eta_denominator",
+        "eff_diphoton_mass     'efficiency vs diphoton mass;       Diphoton mass; efficiency'       diphoton_mass_numerator      diphoton_mass_denominator",
+        "eff_photon_phi        'efficiency vs lead phi;            Photon phi [rad]; efficiency'    photon_phi_numerator         photon_phi_denominator",
+        "eff_photon_subphi     'efficiency vs sublead phi;         Photon subphi [rad]; efficiency' subphoton_phi_numerator      subphoton_phi_denominator",
+        "eff_photonr9          'efficiency vs r9;                  Photon r9; efficiency'           photon_r9_numerator          photon_r9_denominator",
+        "eff_photonhoE         'efficiency vs hoE;                 Photon hoE; efficiency'          photon_hoE_numerator         photon_hoE_denominator",
+        "eff_photonEtaPhi      'Photon phi;                        Photon eta; efficiency'          photon_etaphi_numerator      photon_etaphi_denominator",
+        "eff_photon_subr9      'efficiency vs sublead r9;          Photon subr9; efficiency'        subphoton_r9_numerator       subphoton_r9_denominator",
+        "eff_photon_subhoE     'efficiency vs sublead hoE;         Photon subhoE; efficiency'       subphoton_hoE_numerator      subphoton_hoE_denominator",
+        "eff_photon_subEtaPhi  'Photon sublead phi;                Photon sublead eta; efficiency'  subphoton_etaphi_numerator   subphoton_etaphi_denominator",
+        
+    ),
+    efficiencyProfile = cms.untracked.vstring(
+        "eff_photon_vs_LS 'Photon pt efficiency vs LS; LS' photonVsLS_numerator photonVsLS_denominator"
+    ),
+                                    )
 higgsClient = cms.Sequence(
-    vbfmetClient
-)
+    diphotonEfficiency
+    +vbfmetClient
+    )

--- a/DQMOffline/Trigger/python/HiggsMonitoring_cff.py
+++ b/DQMOffline/Trigger/python/HiggsMonitoring_cff.py
@@ -1,7 +1,8 @@
 import FWCore.ParameterSet.Config as cms
-
+from DQMOffline.Trigger.PhotonMonitor_cff import *
 from DQMOffline.Trigger.VBFMETMonitor_cff import *
 
 higgsMonitorHLT = cms.Sequence(
     higgsinvHLTJetMETmonitoring
-)
+    +higgsHLTDiphotonMonitoring
+    )

--- a/DQMOffline/Trigger/python/PhotonMonitor_cff.py
+++ b/DQMOffline/Trigger/python/PhotonMonitor_cff.py
@@ -69,6 +69,53 @@ exoHLTPhotonmonitoring = cms.Sequence(
 )
 
 
+DiphotonMass90_monitoring = hltPhotonmonitoring.clone()
+DiphotonMass90_monitoring.FolderName = cms.string('HLT/Photon/diphotonMass90/')
+DiphotonMass90_monitoring.numGenericTriggerEventPSet.hltPaths = cms.vstring("HLT_Diphoton30_22_R9Id_OR_IsoCaloId_AND_HE_R9Id_Mass90_v*")
+DiphotonMass90_monitoring.nphotons = cms.uint32(2)
+DiphotonMass90_monitoring.photonSelection = cms.string("(pt > 20 && abs(eta)<1.4442 && hadTowOverEm<0.12 && full5x5_sigmaIetaIeta()<0.015 && full5x5_r9>.5)||(pt > 20 && abs(eta)<2.5 && abs(eta)>1.5556 && hadTowOverEm<0.12 && full5x5_sigmaIetaIeta()<0.035 && full5x5_r9>.8)")
+
+DiphotonMass95_monitoring = hltPhotonmonitoring.clone()
+DiphotonMass95_monitoring.FolderName = cms.string('HLT/Photon/diphotonMass95/')
+DiphotonMass95_monitoring.numGenericTriggerEventPSet.hltPaths = cms.vstring("HLT_Diphoton30_22_R9Id_OR_IsoCaloId_AND_HE_R9Id_Mass95_v*")
+DiphotonMass95_monitoring.nphotons = cms.uint32(2)
+DiphotonMass95_monitoring.photonSelection = cms.string("(pt > 20 && abs(eta)<1.4442 && hadTowOverEm<0.12 && full5x5_sigmaIetaIeta()<0.015 && full5x5_r9>.5)||(pt > 20 && abs(eta)<2.5 && abs(eta)>1.5556 && hadTowOverEm<0.12 && full5x5_sigmaIetaIeta()<0.035 && full5x5_r9>.8)")
+
+DiphotonMass55AND_monitoring = hltPhotonmonitoring.clone()
+DiphotonMass55AND_monitoring.FolderName = cms.string('HLT/Photon/diphotonMass55AND/')
+DiphotonMass55AND_monitoring.numGenericTriggerEventPSet.hltPaths = cms.vstring("HLT_Diphoton30PV_18PV_R9Id_AND_IsoCaloId_AND_HE_R9Id_PixelVeto_Mass55_v*")
+DiphotonMass55AND_monitoring.nphotons = cms.uint32(2)
+DiphotonMass55AND_monitoring.photonSelection = cms.string("(pt > 20 && abs(eta)<1.4442 && hadTowOverEm<0.12 && full5x5_sigmaIetaIeta()<0.015 && full5x5_r9>.5)||(pt > 20 && abs(eta)<2.5 && abs(eta)>1.5556 && hadTowOverEm<0.12 && full5x5_sigmaIetaIeta()<0.035 && full5x5_r9>.8)")
+DiphotonMass55AND_monitoring.histoPSet.massBinning = cms.vdouble(50.,51.,52.,53.,54.,55.,56.,57.,58.,59.,60.,61.,62.,63.,64.,65.,66.,67.,68.,69.,70.,75.,80.,90.,110.,150.)
+
+DiphotonMass55EB_monitoring = hltPhotonmonitoring.clone()
+DiphotonMass55EB_monitoring.FolderName = cms.string('HLT/Photon/diphotonMass55EB/')
+DiphotonMass55EB_monitoring.numGenericTriggerEventPSet.hltPaths = cms.vstring("HLT_Diphoton30EB_18EB_R9Id_OR_IsoCaloId_AND_HE_R9Id_PixelVeto_Mass55_v*")
+DiphotonMass55EB_monitoring.nphotons = cms.uint32(2)
+DiphotonMass55EB_monitoring.photonSelection = cms.string("(pt > 20 && abs(eta)<1.4442 && hadTowOverEm<0.12 && full5x5_sigmaIetaIeta()<0.015 && full5x5_r9>.5)")
+DiphotonMass55EB_monitoring.histoPSet.massBinning = cms.vdouble(50.,51.,52.,53.,54.,55.,56.,57.,58.,59.,60.,61.,62.,63.,64.,65.,66.,67.,68.,69.,70.,75.,80.,90.,110.,150.)
+
+DiphotonMass55ANDnoPV_monitoring = hltPhotonmonitoring.clone()
+DiphotonMass55ANDnoPV_monitoring.FolderName = cms.string('HLT/Photon/diphotonMass55ANDnoPV/')
+DiphotonMass55ANDnoPV_monitoring.numGenericTriggerEventPSet.hltPaths = cms.vstring("HLT_Diphoton30PV_18PV_R9Id_AND_IsoCaloId_AND_HE_R9Id_NoPixelVeto_Mass55_v*")
+DiphotonMass55ANDnoPV_monitoring.nphotons = cms.uint32(2)
+DiphotonMass55ANDnoPV_monitoring.photonSelection = cms.string("(pt > 20 && abs(eta)<1.4442 && hadTowOverEm<0.12 && full5x5_sigmaIetaIeta()<0.015 && full5x5_r9>.5)||(pt > 20 && abs(eta)<2.5 && abs(eta)>1.5556 && hadTowOverEm<0.12 && full5x5_sigmaIetaIeta()<0.035 && full5x5_r9>.8)")
+DiphotonMass55EB_monitoring.histoPSet.massBinning = cms.vdouble(50.,51.,52.,53.,54.,55.,56.,57.,58.,59.,60.,61.,62.,63.,64.,65.,66.,67.,68.,69.,70.,75.,80.,90.,110.,150.)
+
+DiphotonMass55EBnoPV_monitoring = hltPhotonmonitoring.clone()
+DiphotonMass55EBnoPV_monitoring.FolderName = cms.string('HLT/Photon/diphotonMass55EBnoPV/')
+DiphotonMass55EBnoPV_monitoring.numGenericTriggerEventPSet.hltPaths = cms.vstring("HLT_Diphoton30EB_18EB_R9Id_OR_IsoCaloId_AND_HE_R9Id_NoPixelVeto_Mass55_v*")
+DiphotonMass55EBnoPV_monitoring.nphotons = cms.uint32(2)
+DiphotonMass55EBnoPV_monitoring.photonSelection = cms.string("(pt > 20 && abs(eta)<1.4442 && hadTowOverEm<0.12 && full5x5_sigmaIetaIeta()<0.015 && full5x5_r9>.5)")
+DiphotonMass55EBnoPV_monitoring.histoPSet.massBinning = cms.vdouble(50.,51.,52.,53.,54.,55.,56.,57.,58.,59.,60.,61.,62.,63.,64.,65.,66.,67.,68.,69.,70.,75.,80.,90.,110.,150.)
 
 
 
+higgsHLTDiphotonMonitoring = cms.Sequence(
+    DiphotonMass90_monitoring
+    +DiphotonMass95_monitoring
+    +DiphotonMass55AND_monitoring
+    +DiphotonMass55EB_monitoring
+    +DiphotonMass55ANDnoPV_monitoring
+    +DiphotonMass55EBnoPV_monitoring 
+)

--- a/DQMOffline/Trigger/python/PhotonMonitor_cfi.py
+++ b/DQMOffline/Trigger/python/PhotonMonitor_cfi.py
@@ -33,4 +33,3 @@ hltPhotonmonitoring.denGenericTriggerEventPSet.dcsPartitions = cms.vint32 ( 24, 
 hltPhotonmonitoring.denGenericTriggerEventPSet.andOrDcs      = cms.bool( False )
 hltPhotonmonitoring.denGenericTriggerEventPSet.errorReplyDcs = cms.bool( True )
 hltPhotonmonitoring.denGenericTriggerEventPSet.verbosityLevel = cms.uint32(1)
-

--- a/HLTriggerOffline/Higgs/python/hltHiggsValidator_cfi.py
+++ b/HLTriggerOffline/Higgs/python/hltHiggsValidator_cfi.py
@@ -4,7 +4,7 @@ import FWCore.ParameterSet.Config as cms
 hltHiggsValidator = cms.EDAnalyzer("HLTHiggsValidator",
         
     hltProcessName = cms.string("HLT"),
-    analysis       = cms.vstring("HWW", "HZZ", "HZZControlPaths", "MuonJet", "Hgg", "HggControlPaths", "Htaunu", "H2tau", "VBFHbb_0btag", "VBFHbb_1btag", "VBFHbb_2btag",  "ZnnHbb","DoubleHinTaus","HiggsDalitz","X4b","TTHbbej","AHttH","WHToENuBB","MSSMHbb","MSSMHbbmu","VBFHToInv"),
+    analysis       = cms.vstring("HWW", "HZZ", "HZZControlPaths", "MuonJet", "Hgg", "Htaunu", "H2tau", "VBFHbb_0btag", "VBFHbb_1btag", "VBFHbb_2btag",  "ZnnHbb","DoubleHinTaus","HiggsDalitz","X4b","TTHbbej","AHttH","WHToENuBB","MSSMHbb","MSSMHbbmu","VBFHToInv"),
     histDirectory  = cms.string("HLT/Higgs"),
     
     # -- The instance name of the reco::GenParticles collection 
@@ -195,20 +195,13 @@ hltHiggsValidator = cms.EDAnalyzer("HLTHiggsValidator",
         ),  
     Hgg = cms.PSet( 
         hltPathsToCheck = cms.vstring(
-            "HLT_Diphoton30_18_R9Id_OR_IsoCaloId_AND_HE_R9Id_Mass90_v",
-            "HLT_Diphoton30PV_18PV_R9Id_AND_IsoCaloId_AND_HE_R9Id_DoublePixelVeto_Mass55_v",
-            "HLT_Diphoton30_18_Solid_R9Id_AND_IsoCaloId_AND_HE_R9Id_Mass55_v",
-            "HLT_Diphoton30EB_18EB_R9Id_OR_IsoCaloId_AND_HE_R9Id_DoublePixelVeto_Mass55_v",
-        ),
-        recPhotonLabel  = cms.string("photons"),
-        # -- Analysis specific cuts
-        minCandidates = cms.uint32(2), 
-        ),
-    # seperate directory because it needs a different relval    
-    HggControlPaths = cms.PSet( 
-        hltPathsToCheck = cms.vstring(
-            "HLT_Diphoton30_18_R9Id_OR_IsoCaloId_AND_HE_R9Id_DoublePixelSeedMatch_Mass70_v",
-        ),
+            "HLT_Diphoton30_22_R9Id_OR_IsoCaloId_AND_HE_R9Id_Mass90_v",
+            "HLT_Diphoton30_22_R9Id_OR_IsoCaloId_AND_HE_R9Id_Mass95_v",
+            "HLT_Diphoton30PV_18PV_R9Id_AND_IsoCaloId_AND_HE_R9Id_PixelVeto_Mass55_v",
+            "HLT_Diphoton30PV_18PV_R9Id_AND_IsoCaloId_AND_HE_R9Id_NoPixelVeto_Mass55_v",
+            "HLT_Diphoton30EB_18EB_R9Id_OR_IsoCaloId_AND_HE_R9Id_NoPixelVeto_Mass55_v",
+            "HLT_Diphoton30EB_18EB_R9Id_OR_IsoCaloId_AND_HE_R9Id_PixelVeto_Mass55_v",
+            ),
         recPhotonLabel  = cms.string("photons"),
         # -- Analysis specific cuts
         minCandidates = cms.uint32(2), 

